tokio/src/sync/mpsc/list.rs

```diff
@@ -230,6 +230,28 @@ impl<T> fmt::Debug for Tx<T> {
 }
 
 impl<T> Rx<T> {
+    pub(crate) fn is_empty(&self) -> bool {
+        unsafe {
+            let block = self.head.as_ref();
+            !block.has_value(self.index)
+        }
+    }
+
+    pub(crate) fn len(&self, tx: &Tx<T>) -> usize {
+        let tail_position = tx.tail_position.load(Acquire);
+        let tail = tx.block_tail.load(Acquire);
+
+        unsafe {
+            let tail_block = &mut *tail;
+
+            if tail_block.is_closed() {
+                tail_position - self.index - 1
+            } else {
+                tail_position - self.index
+            }
+        }
```

Comment:
Please reduce the scope of this unsafe block. You only need it for the `&mut *tail` operation as far as I can tell.
```Rust
let tail_block = unsafe { &mut *tail };
```